app/services/foreman/plugin.rb

```diff
@@ -396,5 +405,30 @@ def extend_template_helpers_by_module(mod)
       end
       allowed_template_helpers(*(mod.public_instance_methods - Module.public_instance_methods))
     end
+
+    def add_manager_roles(resources, opts)
+      manager = Role.find_by :name => 'Manager'
+      add_permissions(manager, resources, opts) do |resource|
+        Permission.where(:resource_type => resource)
+      end
+    end
+
+    def add_viewer_roles(resources, opts)
+      viewer = Role.find_by :name => 'Viewer'
+      add_permissions(viewer, resources, opts) do |resource|
+        Permission.where(:resource_type => resource).where(Permission.arel_table[:name].matches("view_%"))
+      end
+    end
+
+    def add_permissions(role, resources, opts, &block)
+      to_add = []
+      resources.each do |resource|
```

Comment:
would you mind splitting this method a bit, you have block in block and you call yield in one of them, I'd suggest using 

```ruby
def add_permissions(...)
   to_add = resources.map { |r| new_permissions_for(r) }.inject(&:+)
   role.add_permissions! to_add.map(&:name) if to_add.any?
end

def new_permissions_for(r)
  ...
end
```

I don't think the block is worth it, I'd probably prefer only to pass the condition which can be optional. That would also DRY up `Permission.where(:resource_type => resource)` that you have in add_manager_roles and add_viewer_roles